spark3-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestMergeIntoTable.java

```diff
@@ -232,8 +235,65 @@ public void testSourceFromSetOps() throws NoSuchTableException {
     sql(sqlText, targetName);
     sql("SELECT * FROM %s ORDER BY id, dep", targetName);
     assertEquals("Should have expected rows",
-            ImmutableList.of(row(1, "emp-id-1"), row(2, "emp-id-2")),
-            sql("SELECT * FROM %s ORDER BY id ASC NULLS LAST", targetName));
+           ImmutableList.of(row(1, "emp-id-1"), row(2, "emp-id-2")),
+           sql("SELECT * FROM %s ORDER BY id ASC NULLS LAST", targetName));
+  }
+
+  @Test
+  public void testAmbiguousRowInTarget() throws NoSuchTableException {
```

Comment:
The problem isn't the row in target, it is that two source rows match. Could you update the name to `testMultipleUpdatesForTargetRow`?